File: MedContactCore/DataTransferObjects/DoctorDataDto.cs
using MedContactDb.Entities;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MedContactCore.DataTransferObjects

{
    public class DoctorDataDto 
    {
        public Guid Id { get; set; }
        public Guid? RoleId { get; set; }   
        public Guid? UserId { get; set; }
        public UserDto? User { get; set; }
        public bool? IsBlocked { get; set; }
        public bool? ForDeletion { get; set; }
        public string? SpecNameReserved { get; set; }
        public Guid? SpecialityId { get; set; }
        public SpecialityDto? Speciality { get; set; }
        public List<DayTimeTableDto>? DayTimeTables { get; set; }
        public List<MedDataDto>? MedData { get; set; }
    }
}

File: medcontact-angular/medcontact/src/app/models/doctordatadto.ts
import { Daytimetabledto } from "./daytimetabledto";
import { Meddatadto } from "./meddatadto";
import { Specialitydto } from "./specialitydto";
import { Userdto } from "./userdto";

export class Doctordatadto {
  id?: string;
  roleId?: string;
  userId?: string;
  user?: Userdto;
  isBlocked?: boolean;
  forDeletion?: boolean;
  specNameReserved?: string;
  specialityId?: string;
  speciality?:  Specialitydto;
  dayTimeTables?: Array<Daytimetabledto>;
  medData?: Array<Meddatadto>;
}
